src/Microsoft.Tye.Hosting/Model/Application.cs

```diff
@@ -12,14 +12,19 @@ namespace Microsoft.Tye.Hosting.Model
 {
     public class Application
     {
-        public Application(FileInfo source, Dictionary<string, Service> services, ContainerEngine containerEngine)
+        public Application(string name, FileInfo source, Dictionary<string, Service> services, ContainerEngine containerEngine)
         {
+            Name = name;
             Source = source.FullName;
             ContextDirectory = source.DirectoryName!;
             Services = services;
             ContainerEngine = containerEngine;
         }
 
+        public string ID { get; } = Guid.NewGuid().ToString();
```

Comment:
nit: Uppercase ID? Or Id?

Reply:
Looks like the pattern within Tye is `Id`.  Updated.